src/modules/collection/sagas.ts

```diff
@@ -454,7 +457,62 @@ export function* collectionSaga(legacyBuilderClient: BuilderAPI, client: Builder
         toInitializeItems(items)
       ]
 
-      if (paymentMethod === PaymentMethod.FIAT) {
+      // Handle credits payment
+      if (useCredits && paymentMethod === PaymentMethod.MANA) {
+        // Fetch rarities to get the price
+        yield put(fetchRaritiesRequest())
+        const fetchRaritiesRequestResult: { success: FetchRaritiesSuccessAction; failure: FetchRaritiesFailureAction } = yield race({
+          success: take(FETCH_RARITIES_SUCCESS),
+          failure: take(FETCH_RARITIES_FAILURE)
+        })
+
+        if (fetchRaritiesRequestResult.failure) {
+          throw new Error('Could not fetch rarities: ' + fetchRaritiesRequestResult.failure.payload.error)
+        }
+
+        const rarities = fetchRaritiesRequestResult.success.payload.rarities
+        const rarity = rarities[0]
+
+        if (!rarity || !rarity.prices) {
+          throw new Error('Rarity prices not found')
+        }
+
+        // Get user's credits
+        const creditsResponse: CreditsResponse | null = yield select(getCredits, from)
+
+        if (!creditsResponse || !creditsResponse.credits || creditsResponse.credits.length === 0) {
+          throw new Error('No credits available')
+        }
+
+        // Calculate total price
+        const totalPrice = ethers.BigNumber.from(rarity.prices.MANA).mul(items.length).toString()
+
+        // Get the credits server URL from config
+        const creditsServerUrl: string = yield call([config, config.get], 'CREDITS_SERVER_URL')
```

Comment:
WDYT about instantiating the `CreditsService` where the builder client or the legacy builder client are instantiated and then pass it when building the sagas?